OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseHomeTabFragment.kt

```diff
@@ -292,10 +290,40 @@ class CourseHomeTabFragment : OfflineSupportBaseFragment(), DownloadManagerCallb
 
     override fun download(downloadableVideos: List<HasDownloadEntry>) {
         downloadEntries = downloadableVideos.toMutableList()
-        storagePermissionLauncher.launch(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+            onPermissionGranted()
+        } else {
+            storagePermissionLauncher.launch(Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        }
     }
 
     private fun initVideoObserver() {
+        videoViewModel.refreshUI.observe(viewLifecycleOwner, EventObserver {
+            if (it) {
+                updateListUI()
+            }
+        })
+
+        videoViewModel.infoMessage.observe(viewLifecycleOwner, EventObserver { strResId ->
+            showInfoMessage(getString(strResId))
+        })
+
+        videoViewModel.downloadSizeExceeded.observe(viewLifecycleOwner, EventObserver {
+            if (it.isNotEmpty()) {
+                DownloadSizeExceedDialog.newInstance(object : IDialogCallback {
+                    override fun onPositiveClicked() {
+                        videoViewModel.startDownload(it)
+                    }
+
+                    override fun onNegativeClicked() {
+                        EventBus.getDefault().post(BulkVideosDownloadCancelledEvent())
+                    }
+                }).apply {
+                    show(requireActivity().supportFragmentManager, "dialog")
+                }
```

Comment:
`apply` leads to an app crash move cuz `requireActivity` called for `DownloadSizeExceedDialog` not for the `CourseHomeTabFragment`, so directly call show method like below.

```
                DownloadSizeExceedDialog.newInstance(object : IDialogCallback {
                    override fun onPositiveClicked() {
                        videoViewModel.startDownload(it)
                    }

                    override fun onNegativeClicked() {
                        EventBus.getDefault().post(BulkVideosDownloadCancelledEvent())
                    }
                }).show(childFragmentManager, "dialog")
```